### ISSUE TITLE
Mandatory relay protection partially breaks EVM compatibility

### ISSUE THREAD

**Issue by d10r** [celo-org/celo-blockchain#1734]
I'm checking the feasibility of deploying the Superfluid protocol on Celo.  
So far, everything looks good. Except: Celo seems to have made EIP-155 style relay protection mandatory on the protocol level. That makes it impossible to deploy the protocol as is, because it uses the ERC-777 standard which depends on ERC-1820.  
ERC-1820 however [can be deployed only with a non-replay-protected tx](https://eips.ethereum.org/EIPS/eip-1820#deployment-method). It's one of the use cases where the missing replay protection is actually a feature - it allows deterministic contract addresses across all EVM chains.  
I'm not sure if this breakage was considered when making that change. Maybe you want to re-consider and make mandatory replay protection a default just for the public RPC instead, as e.g. Avalanche is doing.

**Comment by d10r** [comment]
I just did a check of ERC-1820 relevance, on Ethereum mainnet it's used pretty frequently as can be seen here: https://etherscan.io/address/0x1820a4b7618bde71dce8cdc73aab6c95905fad24/advanced#internaltx
Could of course be deployed at a different address on Celo, but that would add considerable friction.

**Comment by saurik** [comment]
Yeah: CELO is simply not compliant if it fails to support Nick's method, so removing support for non-EIP-155 transactions (as described in issue #972) isn't really an option. I failed to see this issue before filing my own #1737, but right after (while adding another issue reference) I ran across this one, and so have closed mine as a duplicate. FWIW, here was my (longer) argument.

> As far as I understand from reading various documentation and issues (including the recent issue #972 and CIP-0035), CELO requires EIP-155 transactions. However, singleton contract deployments must not use EIP-155 :(. It is thereby impossible on CELO to deploy such contracts as the EIP-1820 pseudo-introspection registry, the EIP-2470 singleton factory, or any similar third-party mechanisms (such as Gnosis deployer, or what I do for Orchid).
> 
> I do appreciate the thought process with respect to giving your users some guidance to avoid this attack, but simply banning EIP-155 is too far: there needs to be some kind of escape hatch for developers who know what they are doing, so that we can still use Nick's method to deploy our decentralized "singleton" contracts--some of which being indirectly required by still other standards (such as EIP-777)--at deterministic, cross-chain universal addresses.

**Comment by saurik** [comment]
> Maybe you want to re-consider and make mandatory replay protection a default just for the public RPC instead, as e.g. Avalanche is doing.

FWIW, I also believe that this was egregious on their part. I have provided extensive argument at https://github.com/ethereum/go-ethereum/issues/23152, along with chain analysis, to show that--despite claims to the contrary--no substantial thought was put into this RPC relay decision by the go-ethereum developers (and, also despite claims to the contrary, this decision is in fact clearly being replicated by other chains).

**Comment by YazzyYaz** [comment]
Hi @saurik @d10r we have an All-Core Dev call this week for Celo to discuss including CIP-50 into Espresso HF which will resolve these issues. We invite you to attend and discuss your issues there so we can fast track the CIP to resolve the issue. https://github.com/celo-org/celo-proposals/issues/301
